rust/lance/src/datafusion/logical_plan.rs

```diff
@@ -0,0 +1,147 @@
+use std::{any::Any, sync::Arc};
+
+use arrow_schema::Schema as ArrowSchema;
+use async_trait::async_trait;
+use datafusion::{
+    datasource::TableProvider,
+    error::Result as DatafusionResult,
+    execution::context::SessionState,
+    logical_expr::{LogicalPlan, TableProviderFilterPushDown, TableType},
+    optimizer::utils::conjunction,
+    physical_plan::{filter::FilterExec, limit::GlobalLimitExec, ExecutionPlan, Statistics},
+    prelude::Expr,
+};
+
+use crate::{
+    datafusion::physical_expr::column_names_in_expr,
+    io::exec::{Planner, ProjectionExec},
+    Dataset,
+};
+
+#[async_trait]
+impl TableProvider for Dataset {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> Arc<ArrowSchema> {
+        Arc::new(self.schema().into())
+    }
+
+    fn table_type(&self) -> TableType {
+        TableType::Base
+    }
+
+    fn get_table_definition(&self) -> Option<&str> {
+        None
+    }
+
+    fn get_logical_plan(&self) -> Option<&LogicalPlan> {
+        None
+    }
+
+    async fn scan(
```

Comment:
Originally I did the same as was done here https://github.com/lancedb/lance/pull/1521/files#diff-aa127fed9fd03bbd48ba671543f8fd2991985b373cca51bf620ae9ca264810caR47-R81, but it required a bit of extra processing to go through the scanner. So i refactored it to use the same logic as the scanner. This way avoids unnecessary allocations & compute.